### PR TITLE
Test result bulk upload - error handling improvement

### PR DESCRIPTION
**Pull request by DanielSass**
# BACKEND PULL REQUEST

## Related Issue

- #3644 

## Changes Proposed

- Parse error messages from report stream when a feign exception happens

## Additional Information

- Any 4xx+ http status code results in an exception getting thrown and the response body (which we want) isn't parsed in the usual fashion.

## Testing

- Deploy to any env and submit an invalid csv file (https://reportstream.cdc.gov/assets/csv/ReportStream-StandardCSV-ExampleData-20220509.csv).  Should result in some useful error messages displayed about missing fields or headers.

## Checklist for Primary Reviewer

- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke tested
- [ ] Any content updates (user-facing error messages, etc) have been approved by content team
- [ ] Any changes that might generate questions in the support inbox have been flagged to the support team
- [ ] GraphQL schema changes are backward compatible with older version of the front-end
- [ ] Changes comply with the SimpleReport Style Guide
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java [hunk 1]
```diff
@@ -95,23 +96,24 @@ void ioException_CaughtAndThrowsCsvException() throws IOException {
...
+                      .withBody(x.readAllBytes())));
+    }
     InputStream input = mock(InputStream.class);
     when(input.readAllBytes()).thenReturn(new byte[] {45});
 
     var response = this._service.processResultCSV(input);
 
-    assertEquals(1, response.getErrors().length);
+    assertEquals(6, response.getErrors().length);
     var errorMessage = Arrays.stream(response.getErrors()).findFirst().get();
```
Comment (nathancrtr) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:114]: Looks like this variable is unused, did you want to run an assertion against it?

File: backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java [hunk 2]
```diff
@@ -131,10 +133,8 @@ void feignGeneralError_returnsGenericErrorMessage() throws IOException {
 
     var response = this._service.processResultCSV(input);
 
-    assertEquals(1, response.getErrors().length);
-    var errorMessage = Arrays.stream(response.getErrors()).findFirst().get();
-
-    assertEquals("Server Error", errorMessage.getMessage());
+    assertNull(response.getErrors());
+    assertEquals(UploadStatus.FAILURE, response.getStatus());
```
[the remark this answers is not in the document]
Comment (DanielSass) [reply backend/src/test/java/gov/cdc/usds/simplereport/service/TestResultUploadServiceTest.java:137]: yes good catch

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java [hunk 3]
```diff
@@ -29,6 +31,9 @@ public class TestResultUploadService {
   private final DataHubClient _client;
   private final OrganizationService _orgService;
 
+  private static final ObjectMapper mapper =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:35]: Wow! its crazy this is not default!

**Review by zdeveloper — APPROVED** [review]
LGTM!

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java (hunk 3 above)
Comment (DanielSass) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:35]: yeah it threw me for a loop initially

File: backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java [hunk 4]
```diff
@@ -70,12 +72,23 @@ public TestResultUpload processResultCSV(InputStream csvStream)
               response.getWarnings(),
               response.getErrors());
 
-      _repo.save(result);
+      if (response.getOverallStatus() != ReportStreamStatus.ERROR) {
+        _repo.save(result);
```
Comment (nathancrtr) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java:76]: Good call 👍 this brings us in alignment with how ReportStream's uploader handles this case

**Review by mehansen — APPROVED** [review]

**Review by nathancrtr — APPROVED** [review]
🚀

**Pull request merged**